lib/checkstl.cpp

```diff
@@ -739,13 +759,9 @@ static ValueFlow::Value getLifetimeIteratorValue(const Token* tok, MathLib::bigi
         if (it2 == values.cend())
             return *it;
     }
+    values = pruneLifetimes(values);
```

Comment:
I would call this on the line above at 755: `std::vector<ValueFlow::Value> values = pruneLifetimes(ValueFlow::getLifetimeObjValues(tok, false, path))`